editor.planx.uk/src/pages/FlowEditor/components/EditorMenu.tsx

```diff
@@ -0,0 +1,123 @@
+import FlagIcon from '@mui/icons-material/Flag';
+import FormatListBulletedIcon from '@mui/icons-material/FormatListBulleted';
+import DataObject from '@mui/icons-material/DataObject';
+import Article from '@mui/icons-material/Article';
+import RuleFolderIcon from '@mui/icons-material/RuleFolder';
+import Box from "@mui/material/Box";
+import IconButton from '@mui/material/IconButton';
+import { styled } from "@mui/material/styles";
+import Tooltip, { tooltipClasses,TooltipProps } from '@mui/material/Tooltip';
+import React from "react";
+import { useCurrentRoute, useNavigation } from 'react-navi';
+import { rootFlowPath } from 'routes/utils';
+import { FONT_WEIGHT_SEMI_BOLD } from 'theme';
+
+const MENU_WIDTH = "46px";
+
+const Root = styled(Box)(({ theme }) => ({
+  width: MENU_WIDTH,
+  flexShrink: 0,
+  background: theme.palette.background.paper,
+  borderRight:`1px solid ${theme.palette.border.main}`,
+}));
+
+const MenuWrap = styled("ul")(({ theme }) => ({
+  listStyle: "none",
+  margin: 0,
+  padding: theme.spacing(5, 0, 0, 0),
+}));
+
+const MenuItem = styled("li")(({ theme }) => ({
+  margin: theme.spacing(0.75, 0),
+  padding: 0,
+}));
+
+const TooltipWrap = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} arrow placement="right" classes={{ popper: className }} />
+))(() => ({
+  [`& .${tooltipClasses.arrow}`]: {
+    color: "#2c2c2c",
+  },
+  [`& .${tooltipClasses.tooltip}`]: {
+    backgroundColor: "#2c2c2c",
+    left: "-5px",
+    fontSize: "0.8em",
+    borderRadius: 0,
+    fontWeight: FONT_WEIGHT_SEMI_BOLD,
+  },
+}));
+
+const MenuButton = styled(IconButton, {
+  shouldForwardProp: (prop) => prop !== "isActive",
+})<{ isActive: boolean }>(({ theme, isActive }) => ({
+  color: theme.palette.primary.main,
+  width: MENU_WIDTH,
+  height: MENU_WIDTH,
+  border: "1px solid transparent",
+  borderRightColor: theme.palette.border.main,
+  "&:hover": {
+    background: "white",
+    borderTopColor: theme.palette.border.light,
+    borderBottomColor: theme.palette.border.light,
+  },
+  ...(isActive && {
+    background: theme.palette.common.white,
+    color: theme.palette.text.primary,
+    border: `1px solid ${theme.palette.border.main}`,
+    borderRightColor: "transparent",
+  }),
+}));
+
+function EditorMenu() {
+  const { navigate } = useNavigation();
+  const { lastChunk } = useCurrentRoute();
+  const rootPath = rootFlowPath();
+
+  const isActive = (route: string) => lastChunk.url.pathname.endsWith(route);
+
+  const routes = [
```

Comment:
@ianjon3s This pattern of splitting up "content" and "code" is a good one to be aware of. It's now a bit simpler to add or remove a menu item, or control the visual appearance of all of them consistently 👍